snapcraft/internal/lxd/_containerbuild.py

```diff
@@ -253,6 +254,10 @@ def _inject_snapcraft(self, *, new_container: bool):
         if common.is_snap():
             with tempfile.TemporaryDirectory(
                     prefix='snapcraft', dir=self._lxd_common_dir) as tmp_dir:
+                # Wait for any on-going refreshes to finish.
+                # If there are no changes an error will be returned.
+                with contextlib.suppress(ContainerRunError):
+                    self._container_run(['snap', 'watch', '--last=refresh'])
```

Comment:
It sounds like this only covers manual refreshes. We probably want `--last=auto-refresh` here. I see no problem to just use both if `--last=refresh` helps you test this.

Reply:
Changed.